qiskit/optimization/problems/quadratic_program.py

```diff
@@ -1093,6 +1093,62 @@ def from_ising(self,
         offset -= offset
 
 
+    def is_feasible(self, x: Union[List[float], np.ndarray], detailed = False) -> bool:
+
+        satisfied_variables = []
+        for i, v in enumerate(x):    
+            variable = self.get_variable(i)
+            if variable._lowerbound <= v <= variable._upperbound :
+                satisfied_variables.append(True)
+                # print(f'{variable._name} is within the bounds') 
+            else :
+                satisfied_variables.append(False)
+                # print(f'{variable._name} is outside the bounds')
+
+        satisfied_linear_constraints = []
+        for constraint in self._linear_constraints:
+            lhs = constraint.evaluate(x)
+            if constraint.sense == ConstraintSense.LE:
+                satisfied_linear_constraints.append(lhs <= constraint.rhs)
+            elif constraint.sense == ConstraintSense.GE:
+                satisfied_linear_constraints.append(lhs >= constraint.rhs)
+            elif constraint.sense == ConstraintSense.EQ:
+                satisfied_linear_constraints.append(isclose(lhs, constraint.rhs))
+            else:
+                raise QiskitOptimizationError("Invalid sense!")
+
+        satisfied_quadratic_constraints = []    
+        for constraint in self._quadratic_constraints:
+            lhs = constraint.evaluate(x)
+            if constraint.sense == ConstraintSense.LE:
+                satisfied_quadratic_constraints.append(lhs <= constraint.rhs)
+            elif constraint.sense == ConstraintSense.GE:
+                satisfied_quadratic_constraints.append(lhs >= constraint.rhs)
+            elif constraint.sense == ConstraintSense.EQ:
+                satisfied_quadratic_constraints.append(isclose(lhs, constraint.rhs))
+            else:
+                raise QiskitOptimizationError("Invalid sense!")
```

Comment:
You may collapse two loops into one (if pylint permits) by `for c in self._linear_constraints + self._quadratic_constraints`. Anyway, two mostly identical loops don't look very good.
Also, I think we don't need to raise an exception here.

Reply:
This is same as above, just for debug. I agree that a combined loop is better and cleaner.

Reply:
You may need to add cast to pass pylint as the following code. Please check the pylint result.
https://github.com/Qiskit/qiskit-aqua/blob/408711a65c5491b8c56c9083046df09e9b4333ee/qiskit/optimization/algorithms/slsqp_optimizer.py#L192

Reply:
I checked using pylint as well as using `make style` and `make lint` but I didn't see any errors on the part. Should I add the line regardless for precautions?

Reply:
If pylint and all that make commands do not object, then keep your code. Honestly I don't remember why I added those casts in SLSQP :)

Reply:
Sure, also, could you and @t-imamichi  review the new changes I made so that once finalized, I can do the same changes in the remaining opimizer files :)

Reply:
I see. If the cast is not necessary, it would be good to remove it from SLSQP in another pullreq.

Reply:
I think the cast is necessary because this pr fails mypy as follows.
```
qiskit/optimization/problems/quadratic_program.py:1127: error: Unsupported operand types for + ("List[LinearConstraint]" and "List[QuadraticConstraint]")
qiskit/optimization/problems/quadratic_program.py:1142: error: No overload variant of "list" matches argument type "Callable[[], ValuesView[bool]]"
qiskit/optimization/problems/quadratic_program.py:1142: note: Possible overload variant:
qiskit/optimization/problems/quadratic_program.py:1142: note:     def [_T] __init__(self, iterable: Iterable[_T]) -> List[_T]
qiskit/optimization/problems/quadratic_program.py:1142: note:     <1 more non-matching overload not shown>
```

Reply:
Yeah, probably that's why I've added it.